storage/innobase/mtr/mtr0mtr.cc

```diff
@@ -1223,7 +1223,12 @@ inline void log_t::resize_write(lsn_t lsn, const byte *end, size_t len,
         lsn+= l;
       }
       lsn-= resizing;
-      s= START_OFFSET + lsn % resize_capacity;
+      if (UNIV_UNLIKELY(lsn >= resize_capacity))
+      {
+        lsn-= resize_capacity;
+        resize_lsn.fetch_add(resize_capacity);
+      }
+      s= START_OFFSET + lsn;
```

Comment:
Is it guaranteed that the gap between resize_lsn to lsn is less than 2 x resize_capacity ? If not then please consider having a loop here which should execute once or none in most cases (Also possible to % and / to get to the numbers).

Also, please consider putting an assert like ut_a(lsn < resize_capacity); after we are done advancing the resize_lsn. I think release assert is better here as anything wrong could lead to corruption.

Reply:
I had a similar doubt myself. In fact, the atomic operations on `resize_lsn` may not be safe, because multiple calls can execute concurrently, under shared `log_sys.latch`. I will think of something else, such as acquiring `log_sys.lsn_lock`. Then, `std::memory_order_relaxed` loads and stores should be safe.

Reply:
In 7064ffadc8400836a9a4b8589bf835d394cc03a3 I revised this so that in case a wrap-around is noticed, the rest of `log_t::resize_write()` will be protected by `lock_lsn()`.